src/librustc_typeck/check/mod.rs

```diff
@@ -1655,6 +1655,16 @@ fn check_opaque_for_inheriting_lifetimes(tcx: TyCtxt<'tcx>, def_id: LocalDefId,
 
             r.super_visit_with(self)
         }
+
+        fn visit_const(&mut self, c: &'tcx ty::Const<'tcx>) -> bool {
+            if let ty::ConstKind::Unevaluated(..) = c.val {
+                // FIXME(lazy_normalization_consts) We currenctly don't detect lifetimes within substs
+                // which would violate this check. Even though the particular substitution is not used
+                // within the const, this should still be fixed.
+                return false;
```

Comment:
I updated this FIXME but am not completely certain this is actually correct as is.

Is this statement correct? Especially the "Even though the particular substitution is not used within the const" part?

Reply:
I sort of have no idea what this FIXME is trying to say, yeah. *Why* don't we do this? I guess it's because the check would basically *always* fail for any constant, even though a lot of the constants are like `{22}`. In other words, we really have to normalize the constant (evaluate it) to do this check.